src/lib/fetch.d.ts

```diff
@@ -0,0 +1,129 @@
+/// The fetch API standard can be found at https://fetch.spec.whatwg.org/
+/// Definitions by: Ryan Graham <https://github.com/ryan-codingintrigue>, Kagami Sascha Rosylight <https://github.com/saschanaz>, Robin Van den Broeck <https://github.com/gamesmaxed>
+
+/// <reference path="lib.streams.d.ts" />
+
+interface Window {

```

Comment:
I have always been curious why fetch has been under window - `window.fetch()`. Why not just make it so that `fetch()` works out of the box? Or might be could support both?